doc/debug-system.md

```diff
@@ -0,0 +1,393 @@
+# Overview
+
+This document specifies the processor debug system. The debug system implements run-control debug and bus access functionality according to the [RISC-V Debug Specification 0.13.2](https://github.com/riscv/riscv-debug-spec/raw/4e0bb0fc2d843473db2356623792c6b7603b94d4/riscv-debug-release.pdf). It can be accessed over JTAG (IEEE Std 1149.1-2013).
+
+## Features
+
+- Run-control debug functionality according to the [RISC-V Debug Specification version 0.13.2](https://github.com/riscv/riscv-debug-spec/raw/4e0bb0fc2d843473db2356623792c6b7603b94d4/riscv-debug-release.pdf), which includes all standard debug features: breakpoints, stepping, access to the CPU's GPRs and arbitrary memory locations.
+- Implementation following the Execution Based method as outlined in Section A.2 of the RISC-V Debug Specification. This allows the execution of arbitrary instructions on the core and requires minimal changes to the core.
+- JTAG Debug Transport Module (DTM) according to the RISC-V Debug Specification.
+- System Bus Access with generic 32 or 64 bit bus interface.
+- Support for up to 2^20 harts through one Debug Module
+
+## Description
+
+The debug system described in this document consists of two modules, which are implemented according to the RISC-V Debug Specification: The Debug Module (DM) implemented with a Program Buffer, and a JTAG Debug Transport Module (DTM).
+
+## Compatibility
+
+The debug system is compatible with the RISC-V Debug Specification 0.13.2.
+The specification marks some features as optional.
+The status of these features is visible from the table below.
+
+**Feature**                            | **Support**
+-------------------------------------- | --------------------------------------------------------------------------------------
+External triggers (Section 3.6)        | Not supported at this time.
+Halt groups (Section 3.6)              | Not supported at this time.
+Hardware triggers                      | Not supported at this time. Triggers read 0 according to the specification.
+System Bus Access (SBA) (Section 3.10) | Supported with generic 32 or 64 bit wide interface.
+Authentication                         | Not supported at this time.
+Non-intrusive access to core registers | Not supported at this time. All registers must be accessed through the program buffer.
+
+# Theory of Operations
+
+The functionality of the debug system is implemented according to the RISC-V Debug Specification.
+Refer to this document for more information.
+
+## Block Diagram
+
```

Comment:
The `Debug Transport Module (DTM)` text is cut off on the left side. Is this a github rendering issue?

Reply:
It's more of a "SVG rendering in different browsers with different fonts" issue. I tried to slightly rework the figure to make it more compatible. If that doesn't do the trick we need to go back to PNGs.